### PR TITLE
US-PinHome-02: Create NewsPin.gtmpl template

### PR DESCRIPTION
**Pull request by SaraBoutej**
Create a template for pinned news in the homepage based on the twoColomn.gtmpl template
FireShot Capture 068 - Accueil - http___localhost_8080_portal_intranet_

### CODE REVIEW

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	 	* Created by The eXo Platform SAS
+	 	* Author : eXoPlatform
+	 	* July 29, 2019
+	 	*/
+
+	import javax.jcr.Node;
+	import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
+	import org.exoplatform.wcm.webui.clv.UICLVPortlet;
+	import org.exoplatform.wcm.webui.Utils;
+	import org.exoplatform.services.wcm.core.NodeLocation;
```
Comment (thomasdelhomenie) [inline packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: Please check indedntation.

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+					}
+				  	def viewNode = currentPageData.get(i);
+				 	if(!Utils.isViewable(viewNode)) continue;
+					def isShowTitle = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_TITLE);
+					def isShowDate = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_DATE_CREATED);
+					def isShowLink = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_LINK);
+					def isShowReadmore = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_READMORE);
+
+					def itemLink = uicomponent.getURL(viewNode);
+					uicomponent.setDateTimeFormat("dd/MM/yyyy");
```
Comment (thomasdelhomenie) [inline packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: The date format should be localized, for example it is  MM/dd/YYYY for US. You should probably use LocalDate : https://howtodoinjava.com/java/date-time/localdate-format-example/

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+					def isShowSummary = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_SUMMARY) && itemSummary != null;
+
+					repo = ((ManageableRepository)viewNode.getSession().getRepository()).getConfiguration().getName();
+          			workspace = viewNode.getSession().getWorkspace().getName();
+          			uuid = viewNode.getUUID();
+
+			%>
+
+            <%=uicomponent.addQuickEditDiv("Column2 newsColumn2 media", viewNode)%>
+             <div class="newsFadeSlider">
```
Comment (thomasdelhomenie) [inline packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: We should use the same pattern for css class names. Here we use camel case whereas for news-pin-title we use hyphens. Not sure what is the better, but we should be consistent IMO.

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+					def itemSummaryField = uicomponent.getSummaryField(viewNode);
+					def isShowSummary = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_SUMMARY) && itemSummary != null;
+
+					repo = ((ManageableRepository)viewNode.getSession().getRepository()).getConfiguration().getName();
+          			workspace = viewNode.getSession().getWorkspace().getName();
+          			uuid = viewNode.getUUID();
+
+			%>
+
+            <%=uicomponent.addQuickEditDiv("Column2 newsColumn2 media", viewNode)%>
```
Comment (thomasdelhomenie) [inline packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: What form will be used for edit ?
Actually I am not sure we should add this capability for now. News can be edited in activities, must we be able to edit them in CLV as well ?

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+                      <%
+                      }
+                    }
+                  }
+
+                   if (isShowDate) {
+	                  %><div class="dateAndMail">
+	                  <span class="news-date">
+	                  <i class="uiIconAppcalendar uiIconAppCalendarPortlet uiIconDefaultApp"></i> $itemDateCreated</span>
+	                  <span class="news-owner"><i class="uiIconProfile uiIconDefaultApp"></i>Posted by: $itemOwner</span>
```
Comment (thomasdelhomenie) [inline packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: Text must be internationalized.

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+				<%
+					}
+				%>
+
+			</div> <%
+			}
+		%>
+	</div>
+</div>
+<script>
```
Comment (thomasdelhomenie) [inline packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: We should move this script in a js file, so we can reuse it easily.

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+
+            <%=uicomponent.addQuickEditDiv("Column2 newsColumn2 media", viewNode)%>
+             <div class="newsFadeSlider">
+					<div class="media-content">
+					<div class="news-pin-title">
+						<%
+							if (uicomponent.isViewMode()){
+	                    if (isShowTitle) {
+	                      if (isShowLink) {
+	                        %><div class="heading"><a href="$itemLink" rel="tooltip" data-placement="bottom" title="$itemName">$itemName</a></div><%
```
Comment (thomasdelhomenie) [inline packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: Where does the link go ? Not sure what is the expected behavior (display the activity with the news ?).

File: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl (hunk 5 above)
Comment (SaraBoutej) [reply packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/NewsPin.gtmpl]: Done!

**Pull request closed without merging**